archive_viewer/widgets/archive_search.py

```diff
@@ -147,29 +147,48 @@ def __init__(self, parent: QObject = None) -> None:
         self.layout.addLayout(self.search_layout)
         self.layout.addWidget(self.loading_label)
         self.layout.addWidget(self.results_view)
+        self.insert_button = QPushButton("Add PVs")
+        self.insert_button.clicked.connect(lambda:self.append_PVs_requested.emit(self.selectedPVs()))
+        self.results_view.doubleClicked.connect(lambda:self.append_PVs_requested.emit(self.selectedPVs()))
+        self.layout.addWidget(self.insert_button)
         self.setLayout(self.layout)
 
+    def selectedPVs(self) -> str:
+        """Figure out based on which indexes were selected, the list of PVs (by string name)
+        The user was hoping to insert into the table. Concatenate them into string form i.e.
+        <pv1>, <pv2>, <pv3>"""
+        indices = self.results_view.selectedIndexes()
+        pv_list = ""
+        for index in indices:
+            pv_name = self.results_table_model.results_list[index.row()]
+            pv_list += pv_name + ", "
+        return pv_list[:-2]
+
     def startDragAction(self, supported_actions) -> None:
         """
         The method to be called when a user initiates a drag action for one of the results in the table. The current
         reason for this functionality is the ability to drag a PV name onto a plot to automatically start drawing
         data for that PV
         """
-        indices = self.results_view.selectedIndexes()
-        if len(indices) > 0:
-            index = indices[0]
-            pv_name = self.results_table_model.results_list[index.row()]
-            drag = QDrag(self)
-            mime_data = QMimeData()
-            mime_data.setText(pv_name)
-            drag.setMimeData(mime_data)
-            drag.exec()
+        drag = QDrag(self)
+        mime_data = QMimeData()
+        mime_data.setText(self.selectedPVs())
+        drag.setMimeData(mime_data)
+        drag.exec_()
+
+    def keyPressEvent(self, e: QKeyEvent) -> None:
+        """Special key press tracker, just so that if enter or return is pressed the formula dialog attempts to submit the formula"""
+        if e.key() == Qt.Key_Return or e.key() == Qt.Key_Enter:
+            self.request_archiver_info()
+        return super().keyPressEvent(e)
 
     def request_archiver_info(self) -> None:
         """Send the search request to the archiver appliance based on the search string typed into the text box"""
+        search_text = self.search_box.text()
+        search_text = search_text.replace("?", ".")
```

Comment:
Should we also add `'*'` as a wildcard character?

Reply:
Might as well.